### PR TITLE
Use manual XML to create VM config

### PR DESCRIPTION
**Pull request by keanucz**
I may have over-done it here and probably got quite a bit wrong with ERB so sorry about that

### CODE REVIEW

File: vm.erb [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    <interface type='network'>
+      <mac address='<%= mac %>'/>
+      <source network='default'/>
+      <model type='virtio'/>
+      <boot order='1'/>
+      <address type='pci' domain='0x0000' bus='0x00' slot='0x03' function='0x0'/>
+    </interface>
+    <disk type='file' device='disk'>
+      <driver name='qemu' type='qcow2'/>
+      <source file='/home/scorpio/hdd/test.qcow2'/>
```
Comment (td512) [inline vm.erb]: You may need to rethink this one, the VM QCOW image won't be the same for all machines

Comment (keanucz) [reply vm.erb]: Yeah... we'll need to figure a way to fetch the QCOW image or just remove it flat out

Comment (td512) [reply vm.erb]: Or you could use a combo of the machine's name and IP with `.` replaced with `-`

Comment (keanucz) [reply vm.erb]: alright, will do

**Comment by keanucz** [comment]
there, should be good

File: worker.rb [hunk 2]
```diff
@@ -18,6 +20,17 @@ def run_every(seconds)
     end
 end
 
+def create_vm(name, cpus, memory, volume_capacity, ip)
+    # stolen from here https://github.com/andrewgho/genmac
+    mac = (1..3).collect { "%02x" % [rand 255] }.join(":")
+    mac = "AE:AE:AE:#{mac}"
+    uuid = SecureRandom.uuid
+    vm = ERB.new(File.read("./vm.erb")).result(binding)
+    file = file.open("./vm.conf", "w")
```
Comment (td512) [inline worker.rb]: So you're writing an xml file... to a config file?

File: vm.erb [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    <interface type='network'>
+      <mac address='<%= mac %>'/>
+      <source network='default'/>
+      <model type='virtio'/>
+      <boot order='1'/>
+      <address type='pci' domain='0x0000' bus='0x00' slot='0x03' function='0x0'/>
+    </interface>
+    <disk type='file' device='disk'>
+      <driver name='qemu' type='qcow2'/>
+      <source file='/home/scorpio/hdd/<%= "#{name}-#{ip}" %>'/>
```
Comment (td512) [inline vm.erb]: `#{ip.gsub!(".", "-")}` maybe?

File: worker.rb [hunk 4]
```diff
@@ -43,5 +56,13 @@ def run_every(seconds)
 ensure
 
     conn.close if conn
-    
+
+rescue Fog::Errors::Error => e
```
Comment (td512) [inline worker.rb]: What's this doing here?

**Comment by td512** [comment]
You've tested your code... right?

File: worker.rb [hunk 5]
```diff
@@ -44,7 +44,8 @@ def create_vm(id, name, cpus, memory, volume_capacity, ip, config)
                 puts "Found new pending VM!"
                 puts "Name: #{row["name"]}"
                 mac = create_vm(row["id"], row["name"], row["cpus"], row["memory"], row["volume_capacity"], row["ip"], config)
-                conn.exec("UPDATE #{config["table"]} SET mac=\"#{mac}\" WHERE id=#{row["id"]}")
+                conn.exec("UPDATE #{config["table"]} SET mac=\"#{mac}\" AND status=null WHERE id=#{row["id"]}")
```
Comment (td512) [inline worker.rb]: This isn't the correct way to update multiple columns at the same time. Syntax:

`UPDATE table_name SET expression = value, expression = value WHERE clause`

**Pull request merged**